tensorflow/compiler/xla/service/gpu/gpu_compiler.h

```diff
@@ -64,7 +64,7 @@ class GpuCompiler : public LLVMCompiler {
 
   virtual Status OptimizeHloPostLayoutAssignment(
```

Comment:
Does it need to remain virtual?

Reply:
I think so. If we ever use a base class pointer to point to either `NVPTXCompiler` or `AMDGPUCompiler` we are most likely still want to dispatch to the child classes. This will end up invoking the base class's method if it is non-virtual.